arangod/Cluster/DropCollection.cpp

```diff
@@ -107,6 +108,11 @@ bool DropCollection::first() {
     return false;
   }
 
+
+  // We're removing the shard version from MaintenanceFeature before notifying
+  // for new Maintenance run. This should make sure that the next round does not
+  // get rejected.
+  _feature.delShardVersion(collection);
```

Comment:
if this shard is version 1, the next shard will get version 1 and be rejected.  The logic only works if this shard is something other than version 1.  Either this needs to go away, or be changed to an incShardVersion().  

Then the question becomes what should happen in the fail cases above here.

Reply:
no. if there is nothing the first version == 0